src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Training/TrainingDetails/TrainingDetails.cshtml.cs

```diff
@@ -0,0 +1,65 @@
+#nullable disable
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using Smart.FA.Catalog.Shared.Domain.Enumerations.Training;
+
+namespace Smart.FA.Catalog.Showcase.Web.Pages.Training.TrainingDetails;
+
+public class TrainingDetailsModel : PageModel
+{
+    private readonly Infrastructure.Data.CatalogShowcaseContext _context;
+
+    public TrainingDetailsModel(Infrastructure.Data.CatalogShowcaseContext context)
+    {
+        _context = context;
+    }
+
+    public TrainingDetailsViewModel Training { get; set; }
+
+    public async Task<IActionResult> OnGetAsync(int? id)
+    {
+        if (id == null)
+        {
+            //TO DO: add translated custom messages
+            TempData["errorMessage"] = "There is no record for this request. Please try again.";
+            return RedirectToPage("/404");
+        }
+
+        var trainingDetails = await _context.TrainingDetails.Where(m => m.TrainingId == id).ToListAsync();
+
+        if (!trainingDetails.Any())
+        {
+            //TO DO: add translated custom messages
+            TempData["errorMessage"] = "This training does not exist! Please try again.";
+            return RedirectToPage("/404");
+        }
+
+        Training = MapTrainingDetails(trainingDetails);
+        return Page();
+    }
+
+    private TrainingDetailsViewModel MapTrainingDetails(List<Domain.Models.TrainingDetails> trainingDetails)
```

Comment:
```suggestion
    private TrainingDetailsViewModel MapTrainingDetails(List<Domain.Models.TrainingDetails>? trainingDetails)
```
The following can come later but worth mentionning it.
Having the mapping separated in another class is a good practise too.
Well something like this:
https://gist.github.com/maxime-poulain/ff388bb0e888dcc2e35c0b6fee356582

In this case, it is a bit overkill and might be implemented later on.